src-ezxmlm/decode.ml

```diff
@@ -109,6 +109,16 @@ let data : string decoder =
   match v with `Data s -> Ok s | `El _ -> (fail "Expected Data").dec v
 
 
+let float_decoder : float decoder =
+  Decoder.of_decode_fun
+  @@ fun (v : value) ->
+  match v with
+  | `Data s ->
+      Ok (s |> String.trim |> float_of_string)
+  | `El _ ->
+      (fail "Expected Data").dec v
```

Comment:
* Let's call it `float : float decoder`
* It should use `float_of_string_opt` and `fail` if None (i.e. return `Error` rather than raising an exception if the data is not a valid float)
* It can be implemented using the `data` combinator


```suggestion
let float : float decoder =
  data >>= fun s ->
  match s |> String.trim |> float_of_string_opt with
  | None -> fail "Expected a float"
  | Some f -> succeed f
```

* The build is failing because the implementation for bucklescript is missing. Since `float` above just uses `data` and doesn't care about ezxmlm, you should be able to just copy/paste the same function. You can add it after here: https://github.com/mattjbray/ocaml-decoders/blob/2200d5f6bf81ec7d73c89c76685bb0132e70a5a5/src-bs/bs_xml.ml#L202-L210

Reply:
Thanks, just done! Just waiting on the build to pass.